### PR TITLE
Add EyeLike ONNX operation support

### PR DESCRIPTION
**Pull request by TheDarkchip**
## Summary

Implements support for the EyeLike ONNX operation in burn-import.

## Related Issue

https://github.com/tracel-ai/burn/issues/1714

## Changes

- Added EyeLike operation mapping in burn-import/src/onnx/to_burn.rs
- Implemented eye_like_node function for node conversion  
- Added tests in onnx-tests/tests/eye_like/
- Updated build.rs to include EyeLike test model
- Created ONNX graph using direct construction

## Test Coverage

Tests verify EyeLike operation with:
- Square matrices (3x3)
- Rectangular matrices (3x4)
- Multiple backends (NdArray, WGPU)

The EyeLike operation creates an identity matrix with the same shape as the input tensor.

### CODE REVIEW

File: crates/onnx-ir/src/node/eye_like.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+use crate::ir::{ArgType, Node, TensorType};
+
+/// Create an EyeLike configuration from the node
+/// EyeLike creates an identity matrix with the same shape as the input tensor
+pub fn eye_like_config(_node: &Node) {
+    // EyeLike operation has no configuration parameters
+    // It simply creates an identity matrix with the same shape as the input
```
Comment (antimora) [inline crates/onnx-ir/src/node/eye_like.rs]: Actually, EyeLike op has configurations, please see: https://onnx.ai/onnx/operators/onnx__EyeLike.html#attributes

BTW, there should be an existing op that maps dtype to Burn's onnx type. 

k attribute reminds me of the offset in diag mask https://burn.dev/docs/burn/tensor/struct.Tensor.html#method.diag_mask

Probably you can use the result (bool) into int using `int()`

File: crates/burn-import/src/burn/node/eye_like.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    // Square matrix - use Tensor::eye directly
+                    Tensor::eye(rows, &device)
+                } else {
+                    // Rectangular matrix - create zeros and place eye in top-left
+                    let min_dim = core::cmp::min(rows, cols);
+                    let eye_matrix = Tensor::eye(min_dim, &device);
+                    let result = #input.zeros_like();
+
+                    // Use slice assignment to place the identity matrix in the top-left corner
+                    result.slice_assign([0..min_dim, 0..min_dim], eye_matrix)
```
Comment (antimora) [inline crates/burn-import/src/burn/node/eye_like.rs]: Probably you can use https://burn.dev/docs/burn/tensor/struct.Tensor.html#method.diag_mask instead and convert bools to `int`

**Comment by antimora** [comment]
Thank you for your PR. I did a quick review and added my comments. Let me know if you need more clarification.

File: crates/burn-import/onnx-tests/tests/eye_like/eye_like.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    graph = helper.make_graph(
+        [eye_like_node],
+        'EyeLikeModel',
+        [input_tensor],
+        [output_tensor]
+    )
+
+    # Create model
+    eye_like_model = helper.make_model(graph, producer_name='eye_like_test')
+    eye_like_model.opset_import[0].version = 16
```
Comment (antimora) [inline crates/burn-import/onnx-tests/tests/eye_like/eye_like.py]: There is a way to pass this as a param to make_model

File: crates/burn-import/src/burn/node/eye_like.rs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                }
+            };
+        }
```
[the remark this answers is not in the document]
Comment (TheDarkchip) [reply crates/burn-import/src/burn/node/eye_like.rs]: Runtime assertions is how other ops (matmul) also handle this

**Comment by TheDarkchip** [comment]
## Review Comments Addressed

Thank you @antimora for the thorough review! I've addressed all the main feedback points:

### ✅ EyeLike Attributes Support
- **Added proper `dtype` and `k` attribute parsing** in `EyeLikeConfig`
- **Used existing `element_type_from_proto()` function** for dtype conversion as suggested
- **Implemented k offset support** for upper/lower diagonals

### ✅ Implementation Improvements  
- **Switched to `diag_mask` API** instead of manual tensor operations as recommended
- **Added comprehensive testing** for different k values (-5 to +5) and all supported dtypes
- **Proper type conversion** using `.int()`, `.float()`, `.bool()` based on output tensor type

### ✅ Test Coverage Expansion
- **11 test models** covering edge cases: different dtypes (Float32/64, Int32/64, Bool), various k offsets, boundary conditions (1x1, 1x8, out-of-bounds k values)
- **12 Rust test functions** verifying correctness across multiple scenarios
- **Pure ONNX graph construction** using `helper.make_opsetid()` for proper opset specification

### ✅ Code Quality
- **Proper error handling** with runtime assertions following existing patterns (as used in matmul)
- **Centralized dtype conversion** instead of manual mapping
- **Clean ONNX model generation**

The implementation now fully supports EyeLike's `dtype` and `k` attributes and handles all edge cases properly. All tests pass including the comprehensive edge case coverage.
